Source/WebGPU/WebGPU/Texture.mm

```diff
@@ -2051,12 +2063,17 @@ static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonP
 
     textureDescriptor.sampleCount = descriptor.sampleCount;
 
-    textureDescriptor.storageMode = storageMode(hasUnifiedMemory(), baseCapabilities().supportsNonPrivateDepthStencilTextures);
+    textureDescriptor.storageMode = storageMode(hasUnifiedMemory(), baseCapabilities().supportsNonPrivateDepthStencilTextures, ioSurfaceBacking);
 
     // FIXME(PERFORMANCE): Consider write-combining CPU cache mode.
     // FIXME(PERFORMANCE): Consider implementing hazard tracking ourself.
 
-    id<MTLTexture> texture = [m_device newTextureWithDescriptor:textureDescriptor];
+    id<MTLTexture> texture = nil;
+    if (ioSurfaceBacking)
+        texture = [m_device newTextureWithDescriptor:textureDescriptor iosurface:ioSurfaceBacking plane:0];
```

Comment:
How confident are you that there aren't additional validation rules that are necessary to check before trying to run this line of code? What happens if the texture descriptor indicates we should be trying to make a 3D texture? Or a mipmapped texture? Or a multisample texture?

I don't think this patch needs to handle all these cases, but we at least need a placeholder in the source and a bug filed to determine this stuff.

Reply:
Not very confident - I'll look into what kind of textures can be backed by an `IOSurface` and check for it.

Reply:
I added some validation.